src/main/java/tech/aroma/data/sql/SQLMediaRepository.kt

```diff
@@ -0,0 +1,235 @@
+/*
+ * Copyright 2017 RedRoma, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package tech.aroma.data.sql
+
+import org.slf4j.LoggerFactory
+import org.springframework.dao.EmptyResultDataAccessException
+import org.springframework.jdbc.core.JdbcOperations
+import tech.aroma.data.MediaRepository
+import tech.aroma.data.assertions.RequestAssertions.validImage
+import tech.aroma.data.sql.SQLStatements.*
+import tech.aroma.thrift.Dimension
+import tech.aroma.thrift.Image
+import tech.aroma.thrift.exceptions.*
+import tech.sirwellington.alchemy.arguments.Arguments.checkThat
+import tech.sirwellington.alchemy.arguments.assertions.NumberAssertions.positiveInteger
+import tech.sirwellington.alchemy.arguments.assertions.StringAssertions.validUUID
+import javax.inject.Inject
+
+
+/**
+ *
+ * @author SirWellington
+ */
+internal class SQLMediaRepository
+@Inject constructor(val database: JdbcOperations, val serializer: DatabaseSerializer<Image>) : MediaRepository
+{
+    override fun saveMedia(mediaId: String, image: Image)
+    {
+        checkMediaId(mediaId)
+        checkImage(image)
+
+        val sql = Inserts.MEDIA
+
+        try
+        {
+            database.update(sql,
+                            mediaId.toUUID(),
+                            image.imageType?.toString(),
+                            image.dimension?.width,
+                            image.dimension?.height,
+                            Timestamps.now(),
```

Comment:
Might be better to let the database set this really.
![image](https://cloud.githubusercontent.com/assets/992646/26616528/2fa4adb2-4584-11e7-970e-7885c3ea14ff.png)